utils/build-presets.ini

```diff
@@ -534,6 +534,56 @@ swift-primary-variant-arch=x86_64
 # Don't build the benchmarks
 skip-build-benchmarks
 
+[preset: buildbot_incremental,tools=RA,stdlib=RD,smoketest=macosx,single-thread]
+mixin-preset=buildbot_incremental_base
+build-subdir=buildbot_incremental
+
+# We build release+asserts.
+release
+assertions
+
+# We run the OS X tests and validation tests.
+test
+validation-test
+
+llvm-targets-to-build=X86
+
+# Set the vendor to apple
+compiler-vendor=apple
+
+dash-dash
+
+# Always reconfigure cmake
+reconfigure
+
+# We want to always perform a verbose build
+verbose-build
+
+# Build ninja while we are at it
+build-ninja
+
+# We need to build the unittest extras so we can test
+build-swift-stdlib-unittest-extra
+
+# Make sure our stdlib is RA.
+swift-stdlib-build-type=RelWithDebInfo
+swift-stdlib-enable-assertions=true
+swift-stdlib-use-nonatomic-rc=true
+
+# Disable non-x86 building/testing.
+skip-build-ios
+skip-test-ios
+skip-build-tvos
+skip-test-tvos
+skip-build-watchos
+skip-test-watchos
+stdlib-deployment-targets=macosx-x86_64
+swift-primary-variant-sdk=OSX
+swift-primary-variant-arch=x86_64
+
+# Don't build the benchmarks
+skip-build-benchmarks
+
```

Comment:
@mtake You don't need an incremental build I guess.  I think it could be shorter and more future proof to take `preset: buildbot,tools=RA,stdlib=RD` (based on `mixin_buildbot_tools_RA_stdlib_RD `) as a starting point and slightly change it:

```
[preset: buildbot,tools=RA,stdlib=RD,single-thread]
mixin-preset=mixin_buildbot_tools_RA_stdlib_RD

build-subdir=buildbot_single_thread

dash-dash

# Enable non-atomic build of the stdlib
swift-stdlib-use-nonatomic-rc=true

# Don't run host tests for iOS, tvOS and watchOS platforms to make the build
# faster.
skip-test-ios-host
skip-test-tvos-host
skip-test-watchos-host
```

This way you don't need to repeat a lot of options. You also do not exclude the build of benchmarks.

@gottesmm What do you think? Does it make sense to take `mixin_buildbot_tools_RA_stdlib_RD ` as a basis preset? Or should it be something else?